### PR TITLE
feature/#671 added api for job deletable

### PR DESCRIPTION
**Pull request by toan-quach**
#671

### CODE REVIEW

**Review by jrobinAV — DISMISSED** [review]
Neat.

**Review by trgiangdo — COMMENTED** [review]
The PR looks good.

I only have a few minor comments:
1. Should we expose the `is_deletable()` method at Job level as well? Just like scenario, so the user can do something like `job.is_deletable()`.
2. Should the `test_is_deletable()` be duplicated to the `test_job_manager_with_sql_repo.py` as well?

**Comment by jrobinAV** [comment]
> The PR looks good.
> 
> I only have a few minor comments:
> 
> 1. Should we expose the `is_deletable()` method at Job level as well? Just like scenario, so the user can do something like `job.is_deletable()`.
> 2. Should the `test_is_deletable()` be duplicated to the `test_job_manager_with_sql_repo.py` as well?

Yes, good ideas.

**Comment by toan-quach** [comment]
@trgiangdo you're right! lemme add that as well!

File: tests/core/job/test_job_manager_with_sql_repo.py [hunk 1]
```diff
@@ -207,6 +208,56 @@ def test_force_deleting_unfinished_job(init_sql_repo):
     assert_true_after_time(lambda: reference_last_edit_date != dn_3.last_edit_date)
 
 
+def test_is_deletable():
```
Comment (trgiangdo) [inline tests/core/job/test_job_manager_with_sql_repo.py]:
```suggestion
def test_is_deletable(init_sql_repo):
```

**Review by trgiangdo — DISMISSED** [review]
Just a minor comment to initialize the SQL repo on the test.

File: src/taipy/core/job/job.py [hunk 2]
```diff
@@ -341,3 +341,13 @@ def get_simple_label(self) -> str:
             The simple label of the job as a string.
         """
         return self._get_simple_label()
+
+    def is_deletable(self) -> bool:
+        """Indicate if the scenario can be deleted.
```
Comment (jrobinAV) [inline src/taipy/core/job/job.py]:
```suggestion
        """ Indicate if the job can be deleted.
```

File: src/taipy/core/job/job.py [hunk 3]
```diff
@@ -341,3 +341,13 @@ def get_simple_label(self) -> str:
...
+
+        Returns:
+            True if the scenario can be deleted. False otherwise.
```
Comment (jrobinAV) [inline src/taipy/core/job/job.py]:
```suggestion
            True if the job can be deleted. False otherwise.
```

**Review by trgiangdo — APPROVED** [review]

**Pull request merged**